### PR TITLE
[WIP]: magento/adobe-stock-integration#8: Expand preview image on thumbnail …

### PR DESCRIPTION
**Pull request by sidolov**
…click

### Description (*)

Expand preview image on thumbnail click

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration/issues/8: Expand preview image on thumbnail click

### CODE REVIEW

**Comment by sidolov** [comment]
@passtet , please, take a look, I know that you have an alternative implementation, but we changed our components recently

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image-preview.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        isVisible (record) {
+            return this.visibility()[record._rowIndex] || false;
+        },
+
+        /**
+         * Show image preview
+         *
+         * @param {Number} rowIndex
+         */
+        show: function (rowIndex) {
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image-preview.js:53]: show/close are not the best names combination imho, I'd go with open/close or show/hide

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image.js [hunk 2]
```diff
@@ -39,6 +45,13 @@ define([
                 return record.styles;
             }
             return {};
+        },
+
+        /**
+         * Expand image preview
+         */
+        expandPreview: function (record) {
+            this.previewComponent().show(record._rowIndex);
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image.js:54]: While the preview is shown, the thumbnail should be hidden

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image.js [hunk 3]
```diff
@@ -8,14 +8,20 @@ define([
     'use strict';
 
     return Column.extend({
+        defaults: {
+            modules: {
+                previewComponent: '${ $.parentName }.preview'
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image.js:13]: Image component dependency on preview component makes it impossible to create a masonry grid without preview. Could we reverse the dependency?

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Great job! Just a couple of comments

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image.js (hunk 2 above)
Comment (sidolov) [reply AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image.js:54]: We haven't requirements to hide the thumbnail image during the preview but we can improve this behavior in the scope of the separate task.

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image-preview.js (hunk 1 above)
Comment (sidolov) [reply AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image-preview.js:53]: Agree

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image.js (hunk 3 above)
Comment (sidolov) [reply AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image.js:13]: I'm not sure that we can listen to events from the another component but I'll double-check and make the changes if it's possible

**Pull request merged**